include/cell/compute/math_functor.hpp

```diff
@@ -7,25 +7,130 @@ namespace tiledcuda::cell::compute {
 template <typename Element>
 struct Add {
     DEVICE Element operator()(Element a, Element b) const { return a + b; }
+
+    DEVICE void operator()(const Element& lhs, const Element& rhs,
+                           Element& dst) {
+        dst = lhs + rhs;
+    }
+};
+
+template <typename Element>
+struct Sub {
+    DEVICE Element operator()(Element a, Element b) const { return a - b; }
+
+    DEVICE void operator()(const Element& lhs, const Element& rhs,
```

Comment:
I am curious about why we need to support these two different overloads of the operator `()`?

Reply:
Because I want the functor with the same form of operation to be able to be called recursively:
```cpp
template <typename RegTile>
using BaseTileSub = detail::Binary<RegTile, Sub<typename RegTile::DType>>;
template <typename RegTile>
using RegTileSub =
    detail::Binary<RegTile, BaseTileSub<typename RegTile::DType>>;
```
Otherwise, when executing `RegTileSub`, the operator will be called recursively twice - the outer operator is `BaseTileSub` and the inner operator is `Sub`, while `BaseTileSub` requires the `(lhs, rhs, dst)` format to be called.